processing-chain/src/main/groovy/org/orbisgis/processingchain/PrepareOSM.groovy

```diff
@@ -11,19 +11,19 @@ import org.orbisgis.processmanager.ProcessMapper
  *
  * @return
  */
-public static ProcessMapper OSMPreparationMapper(){
-    def loadInitialData = OSMGISLayers.loadInitialData()
-    def prepareBuildings = OSMGISLayers.prepareBuildings()
-    def prepareRoads = OSMGISLayers.prepareRoads()
-    def prepareRails = OSMGISLayers.prepareRails()
-    def prepareVeget = OSMGISLayers.prepareVeget()
-    def prepareHydro = OSMGISLayers.prepareHydro()
-
-    def transformBuildings = FormattingForAbstractModel.transformBuildings()
-    def transformRoads = FormattingForAbstractModel.transformRoads()
-    def transformRails = FormattingForAbstractModel.transformRails()
-    def transformVeget = FormattingForAbstractModel.transformVeget()
-    def transformHydro = FormattingForAbstractModel.transformHydro()
+public static ProcessMapper PrepareOSMMapper(){
+    def loadInitialData = org.orbisgis.osm.OSMGISLayers.loadInitialData()
+    def prepareBuildings = org.orbisgis.osm.OSMGISLayers.prepareBuildings()
+    def prepareRoads = org.orbisgis.osm.OSMGISLayers.prepareRoads()
+    def prepareRails = org.orbisgis.osm.OSMGISLayers.prepareRails()
+    def prepareVeget = org.orbisgis.osm.OSMGISLayers.prepareVeget()
+    def prepareHydro = org.orbisgis.osm.OSMGISLayers.prepareHydro()
+
+    def transformBuildings = org.orbisgis.osm.FormattingForAbstractModel.transformBuildings()
+    def transformRoads = org.orbisgis.osm.FormattingForAbstractModel.transformRoads()
+    def transformRails = org.orbisgis.osm.FormattingForAbstractModel.transformRails()
+    def transformVeget = org.orbisgis.osm.FormattingForAbstractModel.transformVeget()
+    def transformHydro = org.orbisgis.osm.FormattingForAbstractModel.transformHydro()
```

Comment:
I do not know if you had the opportunity to discuss this point with @gpetit but the idea is finally to map the entire chain (from the creation of the generic table to the creation of the statistics). The processes that are in common between BDTopo and OSM are then included twice, once in the prepareBDTOPO mapper, once in the prepareOSM mapper.
You may find the corresponding mapping the code of the prepareBDTOPO.

Reply:
I did but I thought it would be a good idea to start with the inner OSM mapper anyway, as it has been positively tested.
If you agree on it, once this is validated, I'll start adding the upstream and downstream processes in the mapper to get the requested entire chain. 
So how do I do it ?

Reply:
OK good point. We can merge then and you can make a new request with the full mapper.

Reply:
Did you do a scheme that could be used for a better visual appreciation of your mapper ?

Reply:
OK.

Reply:
Not yet ...